pkg/controllers/provisioningv2/rke2/machinedrain/machinedrain.go

```diff
@@ -41,12 +42,12 @@ func (h *handler) OnChange(key string, machine *capi.Machine) (*capi.Machine, er
 	}
 
 	drain := machine.Annotations[planner.DrainAnnotation]
-	if drain != "" {
-		return h.drain(machine, []byte(drain))
+	if drain != "" && machine.Annotations[planner.DrainDoneAnnotation] != drain {
+		return h.drain(machine, drain)
 	}
 
 	if machine.Annotations[planner.UnCordonAnnotation] != "" {
```

Comment:
Should this check be more explicit?
```suggestion
	if machine.Annotations[planner.UnCordonAnnotation] == drain {
```

Reply:
Added a comment explaining above.